[Term Entry] Lua Tables getn()

badday182 opened the pull request:
<!--
👋 Hi, thanks for sending a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

<!-- Please write a summary of the change, such as which topic(s) and file(s) that you have edited or created. Please also include relevant motivation and context: -->
- Added Table.Getn() term for Lua
- Closes issue #3168 #
### Type of Change

<!--- Please delete or cross off the bullet point(s) that are irrelevant to this PR: -->

- Adding a new entry

### Checklist

<!-- Please check ALL the boxes: -->

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Codecademy/docs?pullRequest=3311) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Codecademy/docs?pullRequest=3311) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Codecademy/docs?pullRequest=3311) it.</sub>

caupolicandiaz commented on content/lua/concepts/tables/terms/getn/getn.md:
```diff
@@ -0,0 +1,83 @@
+CatalogContent:
+  - 'learn-lua'
+  - 'paths/computer-science'
+---
```
Please update the spacing in the entry. There should never be multiple empty lines between content, and there should always be one empty line after a header or code block

caupolicandiaz reviewed (changes requested):
@badday182 I've added a couple of initial comments. Please update the entry, and tag me when you've finished making changes. Thanks

badday182 commented:
@caupolicandiaz  Thank you for your feedback, I finished making changes:
- updated the path and the entry to Luau
- updated the spacing in the entry

caupolicandiaz commented on content/luau/concepts/tables/terms/getn/getn.md:
```diff
@@ -0,0 +1,82 @@
+  - 'paths/computer-science'
+---
+
+**This function has been deprecated and is not recommended for use in new code; use #t instead.**
```
```suggestion
> **Note:** This function has been deprecated. It is recommended that new projects use `#t` instead.
```

caupolicandiaz replied:
move this after the description/summary

caupolicandiaz commented on content/luau/concepts/tables/terms/getn/getn.md:
````diff
@@ -0,0 +1,82 @@
+
+## Example 1
+
+```luau
````
```suggestion
```lua
```

caupolicandiaz commented on content/luau/concepts/tables/terms/getn/getn.md:
````diff
@@ -0,0 +1,82 @@
+
+This code will print the following output:
+
+```lua
````
```suggestion
```shell
```

caupolicandiaz commented on content/luau/concepts/tables/terms/getn/getn.md:
````diff
@@ -0,0 +1,82 @@
+
+This code will print the following output:
+
+```
````
```suggestion
```shell
```

caupolicandiaz commented on content/luau/concepts/tables/terms/getn/getn.md:
````diff
@@ -0,0 +1,82 @@
+
+This code will print the following output:
+
+```
````
```suggestion
```shell
```

caupolicandiaz commented on content/luau/concepts/tables/terms/getn/getn.md:
````diff
@@ -0,0 +1,82 @@
+```lua
+local  table = {1, 2, 3, nil, 5}
+local count = table.getn(table
+print(count) -- Prints 4
````
this output should also be in a separate ```shell block

caupolicandiaz commented on content/luau/concepts/tables/terms/getn/getn.md:
````diff
@@ -0,0 +1,82 @@
+5
+```
+
+## Example 4
````
There should be an empty line after the example header

caupolicandiaz reviewed (changes requested):
@badday182 I've added some additional comments.

badday182 commented:
@caupolicandiaz i've finished making changes:
- moved "Note:" to the end of description/summary
- chenged "luau" to "lua"
- chenged/added ```shell block

badday182 commented:
@caupolicandiaz i've made several corrections:
- changed and moved note about deprecated function to the end of description/summary
- luau block changed to lua block
- used shell block where it's correct

caupolicandiaz reviewed (changes requested):
@badday182 There are a number of additional edits that got mixed in with this PR. There should only be one file here, please revert the other changes. You should be able to use the following syntax-
git checkout main -- filepath

caupolicandiaz reviewed (approved):
@badday182 This is ready for a second review.  However, there continue to be a pair of files that should not be part of this PR

badday182 commented:
@caupolicandiaz 
- no empty line after the example header
- valid filenames other directories

badday182 commented:
@caupolicandiaz 
- no empty line after the example header
- valid filenames other directories

caupolicandiaz reviewed (approved):
@badday182 This is ready for a second review.

KTom101 commented on content/luau/concepts/tables/terms/getn/getn.md:
```diff
@@ -0,0 +1,88 @@
+  - 'paths/computer-science'
+---
+
+The **`.getn()`** function in Luau returns the number of elements in a table by iterating over the table and counting the elements, while ignoring any nil elements. This function is a handy tool for various table-related tasks, including checking the table's length, allocating memory for a table, and determining if a table is empty.
```
The note is supposed to go here with a blockquote (change added below 👇 )
```suggestion
The **`.getn()`** function in Luau returns the number of elements in a table by iterating over the table and counting the elements, while ignoring any nil elements. This function is a handy tool for various table-related tasks, including checking the table's length, allocating memory for a table, and determining if a table is empty.

> **Note:** This function has been deprecated. Applying the `.getn()` function will result in an error. Use `#t` instead.
```

KTom101 reviewed (changes requested):
@badday182 I added changes

badday182 commented:
@KTom101 Thank you for your response, I've added your comments into description

caupolicandiaz commented:
@badday182 This entry is currently failing our formatting test and can't be merged until it is resolved. Often, we take responsibility for running the formatter to update entries. However, we don't have access to your remote. Contributors are asked to use the **Allow edits from maintainers** option. Please update your access settings or run the formatter on your own. Thanks

badday182 commented:
@caupolicandiaz I've given maintainers access to editing

KTom101 commented:
@badday182 Thank you. I was able to format the failing [checks](https://www.codecademy.com/resources/docs/contribution-guide/getting-set-up-on-your-computer#:~:text=There%20are%20continuous%20integration%20(CI)%20workflows) using `npm run format:all`. I will go ahead and merge this.
